src/gsi/cplr_get_wrf_mass_ensperts.f90

```diff
@@ -1705,6 +1740,338 @@ subroutine general_read_wrf_mass2(this,filename,g_ps,g_u,g_v,g_tv,g_rh,g_cwmr,g_
+  !               subroutines convert_netcdf_mass and read_wrf_mass_binary_guess.
+  !               There were substantial differences due to different opinion about what
+  !               to use for surface pressure.  This issue should be resolved by coordinating
+  !               with Ming Hu (ming.hu@noaa.gov).  At the moment, these changes result in
+  !               agreement to single precision between this input method and the guess input
+  !               procedure when the same file is read by both methods.
+  !   2012-03-12  whitaker:  read data on root, distribute with scatterv.
+  !                          remove call to general_reload.
+  !                          simplify, fix memory leaks, reduce memory footprint.
+  !                          use genqsat, remove genqsat2_regional.
+  !                          replace bare 'stop' statements with call stop2(999).
+  !   2017-03-23  Hu      - add code to use hybrid vertical coodinate in WRF MASS core
+  !   2019-04-22  CAPS(C. Tong)   - add direct radar DA option
+  !   input argument list:
+  !
+  !   output argument list:
+  !
+  ! attributes:
+  !   language: f90
+  !   machine:  ibm RS/6000 SP
+  !
+  !$$$ end documentation block
+  
+      use netcdf, only: nf90_nowrite
+      use netcdf, only: nf90_open,nf90_close
+      use netcdf, only: nf90_inq_dimid,nf90_inquire_dimension
+      use netcdf, only: nf90_inq_varid,nf90_inquire_variable,nf90_get_var
+      use kinds, only: r_kind,r_single,i_kind
+      use gridmod, only: nsig
+      use constants, only: zero,one,fv,zero_single
+      use hybrid_ensemble_parameters, only: grd_ens
+      use mpimod, only: mpi_comm_world,ierror,mpi_rtype
+      use netcdf_mod, only: nc_check
+      use directDA_radaruse_mod, only: l_use_cvpqx,cvpqx_pval, cld_nt_updt
+      use directDA_radaruse_mod, only: l_cvpnr,cvpnr_pval
+
+      implicit none
+  !
+  ! Declare passed variables
+      class(get_wrf_mass_ensperts_class), intent(inout) :: this
+      real(r_kind),dimension(grd_ens%lat2,grd_ens%lon2,grd_ens%nsig),intent(out):: &
+                                                    g_qr,g_qs,g_qg,g_qnr,g_w
+      character(255),intent(in):: filename
+  !
+  !   Declare local variables
+      real(r_single),allocatable,dimension(:,:,:):: temp_3d
+      real(r_kind),allocatable,dimension(:,:,:):: gg_qr,gg_qs,gg_qg
+      real(r_kind),allocatable,dimension(:,:,:):: gg_qnr
+      real(r_kind),allocatable,dimension(:,:,:):: gg_w
+      real(r_kind),allocatable,dimension(:):: wrk_fill_2d
+      integer(i_kind),allocatable,dimension(:):: dim,dim_id
+  
+      integer(i_kind):: nx,ny,nz,i,j,k,d_max,file_id,var_id,ndim,mype
+      integer(i_kind):: Time_id,s_n_id,w_e_id,b_t_id,s_n_stag_id,w_e_stag_id,b_t_stag_id
+      integer(i_kind):: Time_len,s_n_len,w_e_len,b_t_len,s_n_stag_len,w_e_stag_len,b_t_stag_len
+      integer(i_kind) iderivative
+  
+      real(r_kind):: deltasigma
```

Comment:
The variable, deltasigma, has been identified as unused.  Please remove unused variable.

Reply:
We removed that variable as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-f4844e36d71565749a9ceb0a3aa45ace24257a80b593048fbc2b5585e0bbca8fR1811